CMakeLists.txt

```diff
@@ -341,24 +367,11 @@ else()
 endif()
 
 set(CMK_POWER7 0)
-if(ARCH)
-  set(CHARM_CPU ${ARCH})
-elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
-  set(CHARM_CPU "x86_64")
-elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "i386" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "i686")
-  set(CHARM_CPU "i386")
-elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "powerpc" OR CMAKE_SYSTEM_PROCESSOR MATCHES "^ppc")
-  set(CHARM_CPU "ppc64le")
+if(CHARM_CPU STREQUAL "ppc64le")
   execute_process(COMMAND grep f Makefile RESULT_VARIABLE isppc7 OUTPUT_QUIET ERROR_QUIET)
   if(${isppc7})
     set(CMK_POWER7 1)
   endif()
```

Comment:
The whole POWER 7 stuff probably should be removed. `CMK_POWER7` isn't user anywhere, and we do not support Power 7 anymore I think (we only support ppc64le, which was introduced with Power8)